controllers/envoy/envoyconfig_controller.go

```diff
@@ -21,25 +21,23 @@ import (
 	"fmt"
 
 	envoyv1alpha1 "github.com/3scale/marin3r/apis/envoy/v1alpha1"
-	xdss "github.com/3scale/marin3r/pkg/discoveryservice/xdss"
 
 	"github.com/go-logr/logr"
 	"github.com/operator-framework/operator-lib/status"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
 // EnvoyConfigReconciler reconciles a EnvoyConfig object
 type EnvoyConfigReconciler struct {
-	Client   client.Client
-	Log      logr.Logger
-	Scheme   *runtime.Scheme
-	XdsCache xdss.Cache
```

Comment:
The finalizer has been moved to the EnvoyConfigRevision controler, so access to the xDS server in-memory cache is no longer required for the EnvoyConfig controller.